### ISSUE TITLE
bug: cant customize lsp per project with rustaceanvim 

### ISSUE THREAD

**Issue by Alt-iOS** [LazyVim/LazyVim#2690]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.95

### Operating system/version

macos 14

### Describe the bug

Cant change rust-analyzer settings per porject, and I think its because the default config LazyVim uses

### Steps To Reproduce

1. update to the latest version
2. try to customize in a project the rust-analyzer.json (having marked it as project root)

### Expected Behavior

Rustaceanvim picks up the config and changes based on that. I believe this config provided by the LazyExtra:
<img width="634" alt="LazyVim default config" src="https://github.com/LazyVim/LazyVim/assets/80638470/8049f233-a899-450b-843f-b4999005dcb6">
is colliding with this 
<img width="955" alt="Default rustaceanvim config" src="https://github.com/LazyVim/LazyVim/assets/80638470/d04752ea-d6e3-4037-9ce0-dd806d2b2075">

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/LazyVim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by folke** [comment]
I don't use rust, so feel free to make a PR

**Comment by Alt-iOS** [comment]
I am really new to nvim but if i manage to figure it out, will do

**Comment by dpetka2001** [comment]
Isn't this supposed to be fixed by #2720?

**Comment by folke** [comment]
yep, you're right!
